src/window/resource_settings.c

```diff
@@ -55,6 +55,11 @@ static void init(resource_type resource)
 
 static void draw_background(void)
 {
+    // added this draw call to fix a bug where the Help window is larger than the
+    // trade dialog itself, causing bits of it to remain on the screen when going
+    // back to the dialog. Bug was not present in original C3.
+    window_advisors_draw_dialog_background();
```

Comment:
Unsing `window_draw_underlying_window` here should fix your background issues without requiring hacky callbacks.